src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,57 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    public static final int MAX_RANDOM_PROPERTIES = 50;
+
+    public Figure getRandomFigure() {
+        int randomFigureValue = new Random().nextInt(NameOfFigure.values().length);
+        String figureName = NameOfFigure.values()[randomFigureValue].toString();
```

Comment:
its mentioned in your checkstyle, we should use name not toString on enums
```suggestion
        String figureName = NameOfFigure.values()[randomFigureValue].name();
```